### PR TITLE
Optimise webnfc NFCReader option tests

### PR DESCRIPTION
**Pull request by xiuqijix**
- Add the checkpoint of invalid `recordType` option.
- Use common function to test `url/recordType` option.
- Remove `mediaType/url` in `readOptions`.
- Modify the `else if` structure in nfc_help.js `assertWebNDEFMessagesEqual` function.

### CODE REVIEW

File: web-nfc/NFCReader_options_recordType_empty-manual.https.html [hunk 1]
```diff
@@ -22,13 +22,10 @@
 setup({ explicit_timeout: true });
 
 const desc = "Test that write and read data succeed when recordType is set to 'empty'.";
```
Comment (Honry) [inline web-nfc/NFCReader_options_recordType_empty-manual.https.html]: s/recordType/NFCReaderOptions's recordType

Comment (Honry) [reply web-nfc/NFCReader_options_recordType_empty-manual.https.html]: Please also update the rest.

File: web-nfc/NFCReader_options_recordType_empty-manual.https.html [hunk 2]
```diff
@@ -22,13 +22,10 @@
...
-const readOptions = {
-  recordType: "empty",
-  mediaType: "",
-  url: ""
-}
+const readOptions = {recordType: "empty"};
+const invalidReadOptions = {recordType: "json"};
```
Comment (Honry) [inline web-nfc/NFCReader_options_recordType_empty-manual.https.html]: "invalidReadOptions" sounds odd, as it is not an invalid one.
How about "unacceptableReadOptions"?

File: web-nfc/resources/nfc_help.js [hunk 3]
```diff
@@ -59,21 +60,25 @@ function assertWebNDEFMessagesEqual(a, b) {
...
-    if (typeof recordA.data === 'number'
+    } else if (typeof recordA.data === 'number'
         || typeof recordA.data === 'string') {
       assert_true(recordA.data == recordB.data);
     }
   }
 }
 
-function testNDEFMessage(pushedMessage, readOptions, desc) {
+function testNDEFMessage(pushedMessage, readOptions, invalidReadOptions, desc) {
```
Comment (Honry) [inline web-nfc/resources/nfc_help.js]: Moreover, "testNDEFMessage" is also not good. How about "testNFCReaderOptions"?

**Comment by kaixinjxq** [comment]
@Honry, thank you for your review! I just submit a commit to address your comments, PTAL.

File: web-nfc/NFCReader_options_recordType_json-manual.https.html [hunk 4]
```diff
@@ -21,14 +21,11 @@
...
 setup({ explicit_timeout: true });
 
-const desc = "Test that write and read data succeed when recordType is set to 'json'.";
-const readOptions = {
-  recordType : "json",
-  mediaType : "application/json",
-  url : ""
-}
+const desc = "Test that write and read data succeed when NFCReaderOptions's recordType is set to 'json'.";
+const readOptions = {recordType : "json"};
```
Comment (Honry) [inline web-nfc/NFCReader_options_recordType_json-manual.https.html]: Nit: remove the white space before the colon.

File: web-nfc/NFCReader_options_recordType_opaque-manual.https.html [hunk 5]
```diff
@@ -21,14 +21,11 @@
...
 setup({ explicit_timeout: true });
 
-const desc = "Test that write and read data succeed when recordType is set to 'opaque'."
-const readOptions = {
-  recordType : "opaque",
-  mediaType : "application/octet-stream",
-  url : ""
-}
+const desc = "Test that write and read data succeed when NFCReaderOptions's recordType is set to 'opaque'.";
+const readOptions = {recordType : "opaque"};
```
Comment (Honry) [inline web-nfc/NFCReader_options_recordType_opaque-manual.https.html]: And here.

File: web-nfc/NFCReader_options_recordType_text-manual.https.html [hunk 6]
```diff
@@ -21,14 +21,11 @@
...
 setup({ explicit_timeout: true });
 
-const desc = "Test that write and read data succeed when recordType is set to 'text'."
-const readOptions = {
-  recordType : "text",
-  mediaType : "text/plain",
-  url : ""
-}
+const desc = "Test that write and read data succeed when NFCReaderOptions's recordType is set to 'text'.";
+const readOptions = {recordType : "text"};
```
Comment (Honry) [inline web-nfc/NFCReader_options_recordType_text-manual.https.html]: And here.

**Review by Honry — APPROVED** [review]
LGTM

**Pull request merged**